pkg/util/prometheushelper/helper.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package prometheushelper
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const module = "prometheus_helper"
+
+var (
+	counterMap   = make(map[string]*prometheus.CounterVec)
+	gaugeMap     = make(map[string]*prometheus.GaugeVec)
+	histogramMap = make(map[string]*prometheus.HistogramVec)
+	summaryMap   = make(map[string]*prometheus.SummaryVec)
+)
+
+var (
+	validMetric = regexp.MustCompile(`^[a-zA-Z_:][a-zA-Z0-9_:]*$`)
+	validLabel  = regexp.MustCompile(`^[a-zA-Z_][a-zA-Z0-9_]*$`)
+)
+
+// NewCounter create the counter metric
+func NewCounter(metric string, help string, labels []string) *prometheus.
+	CounterVec {
```

Comment:
per https://github.com/golang/go/wiki/CodeReviewComments#line-length, better not wrap this line, and there are other cases.
```suggestion
func NewCounter(metric string, help string, labels []string) *prometheus. CounterVec {
```